app/code/core/Mage/GoogleAnalytics/Helper/Data.php

```diff
@@ -172,17 +172,23 @@ public function isUserIdEnabled($store = null)
      * Returns last category name
      *
      * @param Mage_Catalog_Model_Product $product
-     * @return string
+     * @return ?string
      */
-    public function getLastCategoryName($product): string
+    public function getLastCategoryName($product): ?string
     {
-        $_categoryIds = $product->getCategoryIds();
-        if ($_categoryIds) {
-            $_lastCat = array_pop($_categoryIds);
-            $_cat = Mage::getModel('catalog/category')->load($_lastCat);
-            return $_cat->getName();
-        }
-        return '';
+        // force root category to current store
+        $rootCategory = Mage::getModel('catalog/category')
+            ->load(Mage::app()->getStore()->getRootCategoryId());
+
+        $_lastCat = Mage::getResourceModel('catalog/category_collection')
+            ->addAttributeToSelect('name')
+            ->addIdFilter($product->getCategoryIds())
+            ->addIsActiveFilter()
+            ->addFieldToFilter('path', array('like' => $rootCategory->getPath() . '/%'))
+            ->addOrder('level')
+            ->getFirstItem();
+
+        return $_lastCat->getName() ?: false;
```

Comment:
```suggestion
        return $_lastCat->getName() ?: '';
```

since the method is called getLastCategoryName I'd expect it to return a string, or a null in the worst case?

Reply:
I'd prefer an empty string. I guess in the end it would be used as "empty string" later anyways?

Reply:
I returned false because in other parts it is tracking false in the datalayer, but it's the same.